### DISCUSSION TITLE
Interface name not appearing in exception message?

### DISCUSSION THREAD

**Discussion by adnang** [reactiveui/refit#1396] [Q&A] [answered]
I've got a method that calls multiple refit client interfaces in sequence. When one of them fails, i don't get any information in the `ApiException` about which of the interfaces failed.

In production, the exception log's line number is the end of the method block, so i can't point to a specific part of the code either:

``` ---> Refit.ApiException: Response status code does not indicate success: 500 (Internal Server Error).
   at Refit.RequestBuilderImplementation.<>c__DisplayClass14_0`2.<<BuildCancellableTaskFuncForMethod>b__0>d.MoveNext() in /_/Refit/RequestBuilderImplementation.cs:line 313
--- End of stack trace from previous location ---
   at Namespace.Class.Method(Request request, CancellationToken ct) in /sln/Namespace/Class.cs:line 109```

Is there a way to expose the interface name to the ApiException?

**Comment by glennawatson** [comment, marked as answer]
Refit does not put the interface or method name into the ApiException message, but the exception does carry enough to identify the failing call. ApiException (via ApiExceptionBase) exposes:

- HttpMethod - the verb
- Uri - the full request URI
- RequestMessage - the underlying HttpRequestMessage

So in a catch you can log which call failed:

```csharp
catch (ApiException ex)
{
    logger.LogError(ex, "Refit call failed: {Method} {Uri}", ex.HttpMethod, ex.Uri);
}
```

If you want the interface/method name specifically, the clean way is a DelegatingHandler that stamps an identifier onto the request (e.g. via HttpRequestMessage.Options or a header) so it shows up on RequestMessage when you catch the exception. There is no built-in field for the interface name today.
